lib/features/give/bloc/give/give_bloc.dart

```diff
@@ -0,0 +1,150 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:developer';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:givt_app/features/give/models/models.dart';
+import 'package:givt_app/features/give/repositories/campaign_repository.dart';
+import 'package:givt_app/shared/repositories/givt_repository.dart';
+import 'package:intl/intl.dart';
+
+part 'give_event.dart';
+part 'give_state.dart';
+
+class GiveBloc extends Bloc<GiveEvent, GiveState> {
+  GiveBloc(
+    this._campaignRepository,
+    this._givtRepository,
+  ) : super(const GiveState()) {
+    on<GiveQRCodeScanned>(_qrCodeScanned);
+
+    on<GiveAmountChanged>(_amountChanged);
+
+    on<GiveOrganisationSelected>(_organisationSelected);
+  }
+
+  final CampaignRepository _campaignRepository;
+  final GivtRepository _givtRepository;
+
+  FutureOr<void> _qrCodeScanned(
+    GiveQRCodeScanned event,
+    Emitter<GiveState> emit,
+  ) async {
+    emit(state.copyWith(status: GiveStatus.loading));
+    try {
+      final uri = Uri.parse(event.rawValue);
+      final mediumId = utf8.decode(base64.decode(uri.queryParameters['code']!));
+      final organisation = await _getOrganisation(mediumId);
+      final transactionList = _createTransationList(mediumId, event.userGUID);
+      //   'advertisementImageUrl': '',
```

Comment:
This comment block can be removed, right? 
Its also as code in a other file :)

Reply:
#18 is addressing this